transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala

```diff
@@ -3,15 +3,38 @@ package org.broadinstitute.monster.encode.transformation
 import java.time.{LocalDate, OffsetDateTime}
 
 import org.broadinstitute.monster.encode.jadeschema.table.Biosample
+import org.slf4j.LoggerFactory
 import upack.Msg
 
 /** Transformation logic for ENCODE biosample objects. */
 object BiosampleTransformations {
   import org.broadinstitute.monster.common.msg.MsgOps
 
+  private val logger = LoggerFactory.getLogger(getClass)
+
   /** Transform a raw ENCODE biosample into our preferred schema. */
-  def transformBiosample(biosampleInput: Msg): Biosample = {
+  def transformBiosample(biosampleInput: Msg, joinedLibraries: Iterable[Msg]): Biosample = {
+    val biosample_id = CommonTransformations.readId(biosampleInput)
     val (auditLevel, auditLabels) = CommonTransformations.summarizeAudits(biosampleInput)
+
+    val product_ids = joinedLibraries
+      .flatMap(library => library.tryRead[String]("product_id"))
+      .toSet[String]
+    val lot_ids = joinedLibraries
+      .flatMap(library => library.tryRead[String]("lot_id"))
+      .toSet[String]
+
+    if (product_ids.size > 1) {
+      logger.warn(
+        s"Biosample '$biosample_id' has more than one product id in: [${product_ids.mkString(",")}]."
+      )
+    }
+    if (product_ids.size > 1) {
+      logger.warn(
+        s"Biosample '$biosample_id' has more than one lot id in: [${lot_ids.mkString(",")}]."
+      )
+    }
```

Comment:
Did you see this in any of our test inputs? We can run it on the full extraction if not, just to see if it ever happens

Reply:
Yeah, it seems like every library associated with a given biosample has the same values for product and lot id. I checked that this holds true in the test cases and the other examples I looked through by hand, but I agree, it would be good to run it on the full extraction and see what happens.